### DISCUSSION TITLE
Consolidate/organize dependency management, VS code plugin, devcontainer, etc.?

### DISCUSSION THREAD

**Discussion by StevenClontz** [PreTeXtBook/pretext-cli#467] [Ideas]
We now have organically spread out authoring toolchains across a few places:

- here
- https://github.com/PreTeXtBook/pretext-codespace
- https://github.com/oscarlevin/pretext-tools

I think we should discuss separation of concerns. I'm also wondering if CLI+pretext-tools should share a repo (if not, pretext-tools should probably be moved to the PreTeXtBook org).

**Comment by bjones1** [comment]
IMHO, PreTeXt-tools should be separate from the CLI, since they target different platforms (desktop app vs VSCode plugin) and have (AFAIK) independent release cycles. I agree that the PreTeXt-tools should be migrated to the PreTeXt gihhub account.

I don't know re: the codespace. I'd suggest getting it to a more mature point, where we've answered some basic questions (how to distribute; how to reduce bootup time; how to deal with free account disk space limitations; etc.) before making this decision.

**Comment by oscarlevin** [comment]
Definitely distinct repos, but I'm not opposed to having it live inside pretextbook's org.  At some point, we should probably figure out how to change the "publisher" in microsoft's extension marketplace from "oscarlevin" to "pretextbook" too.

That said, a topic for discussion, presumably at a workshop about organizing the open source ecosystem or the like, might be what makes the project more inviting and relevant looking.  Is it one "organization" that controls everything, or is it many different individuals and organizations that all have their own projects using and helping pretext.  For example, it is even now possible for someone to create a new `pretext new` template that can be called, and host it at their own repository.  If they did so, I think that makes it appear like pretext has wider reach.

**Comment by StevenClontz** [comment]
I don't think we got to exactly this point last week, but I think moving pretext-tools to PreTeXtBook is a good move. At the very least, it'd be easier for me to find and contribute to!

I think that mission-critical stuff like pretext-cli and pretext-tools belong in a central organization. But ensuring things are wired so folks can contribute via their personal repos is good as well.
